lib/Importer/ONNX.cpp

```diff
@@ -243,29 +243,58 @@ bool ONNXModelLoader::loadOperator(const ONNX_NAMESPACE::NodeProto &op) {
     // Pads : {pad_top, pad_left, pad_bottom, pad_right}
     std::vector<unsigned_t> pads = getPads(dict);
 
+    // Load the inputs:
     auto in = getNodeValueOrCreateVariableByName(op.input(0));
-    Tensor *w = getTensorByName(op.input(1));
-
-    // Transpose the weights to the right format. Glow expects to read the
-    // weights in the format CRSK. ONNX stores the operators as KCRS.
-    // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
-    Tensor wtag;
-    w->transpose(&wtag, NCHW2NHWC);
+    Node *filterTransposeNode = NULL;
+    std::vector<unsigned_t> kernelShape(2);
+    size_t depth;
+
+    if (hasTensorWithName(op.input(1))) {
```

Comment:
maybe isStaticWeight or something like that?

Reply:
I was just saying that we'd need a better name for this method which reflects its intent. If there is a way to avoid this call I'm all for it.

Reply:
@artemrakhov The fact that it's a static weight is used in other places already -- e.g. in the Conv Caffe2 importer, `getTensorByName()` is used to get the filter, and always transposes it immediately before creating a Variable for it. I agree though that we shouldn't do the transpose there, and instead should create a Transpose node, as the optimizer will do this static transposing anyway in `optimizeTranspose()`. And as far as I know (?) there's also no guarantee that the filter is statically known/loaded -- if so then it's theoretically possible we could load a proto that this would break, where the filter is calculated from another node.

However, it is important to understand if a tensor is static in other places. E.g. in `loadReshape()` we use `getTensorByName()` to load such a static tensor to use as the Reshape's shape, and so it's important that this tensor is static.

So I agree with @rdzhabarov on the renaming -- I think at least `getTensorByName()` should have its name changed, but whatever it ends up being named, it and `hasTensorWithName()` should match.

Reply:
Agreed on everything. That is what I meant -- we should add a transpose Node here and in the C2 loader as well, instead of doing the transpose directly on the static tensor. It would fix this potential issue.